src/calculator/calculator.controller.ts

```diff
@@ -0,0 +1,66 @@
+import { Body, Controller, Post } from '@nestjs/common';
+import { ApiBody, ApiOkResponse, ApiOperation, ApiTags } from '@nestjs/swagger';
+import { CalculatorSupportEmail } from 'emails';
+
+import { Mail } from '@/mail/interfaces/mail.interface';
+import { MailService } from '@/mail/mail.service';
+
+import { CalculatorService } from './calculator.service';
+import { ResultDto, SupportDto } from './dtos';
+
+@ApiTags('calculator')
+@Controller({ path: 'calculator', version: '1' })
+export class CalculatorController {
+  constructor(
+    private readonly calculatorService: CalculatorService,
+    private readonly mailService: MailService,
+  ) {}
+
+  @Post('contact')
+  @ApiOperation({
+    summary: 'Request support',
+    description: 'Returns contact information',
+  })
+  @ApiBody({ type: SupportDto })
+  @ApiOkResponse({
+    description: 'Returns created email response',
+  })
+  async contact(@Body() supportDto: SupportDto) {
+    await this.calculatorService.saveContactInfo(supportDto);
+
+    const mail: Mail = {
+      to: supportDto.email,
+      from: 'no-reply@app.recy.life',
+      subject: 'Contact Information',
+      react: CalculatorSupportEmail({
+        companyType: supportDto.companyType,
+        employeesQuantity: supportDto.employeesQuantity,
+        wasteFootPrint: supportDto.wasteFootPrint,
+      }),
+    };
+
+    try {
+      const { data, error } = await this.mailService.sendEmail(mail);
```

Comment:
Allowing mail sending through the front-end might expose an attack vector, where attachers might use the front-end to dispatch mail messages. Is it possible to add some backend check on e-mail requests or even sending e-mails through the backend?

Reply:
Here is our backend @faustovanin.